File: mobile/lib/features/profile/data/models/volunteer_profile_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'volunteer_profile_model.freezed.dart';
part 'volunteer_profile_model.g.dart';

/// Mirrors backend `VolunteerProfileDto`.
@freezed
class VolunteerProfileModel with _$VolunteerProfileModel {
  const factory VolunteerProfileModel({
    required String userId,
    required List<String> skills,
    required String availability,
    required List<String> causes,
    required String location,
    required String bio,
  }) = _VolunteerProfileModel;

  factory VolunteerProfileModel.fromJson(Map<String, dynamic> json) =>
      _$VolunteerProfileModelFromJson(json);
}

File: frontend/src/app/features/profiles/models/volunteer-profile.model.ts
/**
 * Mirrors `VolunteerProfileDto` returned by
 * `GET /api/volunteer-profiles/me` (camelCase via ASP.NET JSON).
 */
export interface VolunteerProfile {
  readonly userId: string;
  readonly skills: readonly string[];
  readonly availability: string;
  readonly causes: readonly string[];
  readonly location: string;
  readonly bio: string;
}

export interface VolunteerProfilePayload {
  readonly skills: readonly string[];
  readonly availability: string;
  readonly causes: readonly string[];
  readonly location: string;
  readonly bio: string;
}
